src/api/auth.js

```diff
@@ -76,12 +73,21 @@ class AuthInterface {
   */
   setupUri() {
     const appInfo = this.app.appInfo;
-    const schema = `safe-${urlsafeBase64(appInfo.id).toLowerCase()}`;
-    lib.registerUriScheme({ bundle: appInfo.id,
-      vendor: appInfo.vendor,
-      name: appInfo.name,
-      icon: 'test',
-      exec: appInfo.customAuthExecPath }, schema);
+    const opts = this.app.options;
+    let scheme;
+    if (opts.registerScheme) {
+      scheme = `safe-${urlsafeBase64(appInfo.id).toLowerCase()}`;
+    }
+    if (opts.joinSchemes && opts.joinSchemes.length > 0) {
+      scheme = scheme ? [scheme].concat(opts.joinSchemes) : opts.joinSchemes;
+    }
+    if (scheme) {
+      lib.registerUriScheme({ bundle: appInfo.id,
```

Comment:
@bochaco I think we'll need the `dev` exclusion here for OSX to pass in the electron bundle identifier.

Reply:
Thanks @joshuef !
As agreed, I'll add a change to support a `bundle` field in the `appInfo`, and it fall backs to `appInfo.id` if not provided.